sdl2promela/stop_condition/translator.py

```diff
@@ -590,6 +685,52 @@ def _resolve_remaining_element_of_selector(
         return memberType
 
 
+def _find_type_of_queue_last_call(context: GenerateContext, call: model.CallExpression):
+    if len(call.parameters) != 1:
+        raise TranslateException("Invalid parameters for 'queue_last'")
+    if not isinstance(call.parameters[0], model.Selector):
+        raise TranslateException(
+            "Invalid parameter for 'queue_last', required <process>.<signal>"
+        )
+    selector = call.parameters[0]
+    if not isinstance(selector.elements[0], model.VariableReference):
+        raise TranslateException(
+            "Invalid parameter for 'queue_last', required <process>.<signal>"
+        )
+    if not isinstance(selector.elements[1], model.VariableReference):
+        raise TranslateException(
+            "Invalid parameter for 'queue_last', required <process>.<signal>"
+        )
+
+    process_name = selector.elements[0].name.lower()
+    if process_name not in context.processes:
+        raise TranslateException(
+            "Cannot find process with name '{}'".format(process_name)
+        )
+    process = context.processes[process_name]
+    signal_name = selector.elements[1].name.lower()
+    found_signal_types = [
+        signal["type"]
+        for signal in process.input_signals
+        if signal["name"] == signal_name
+    ]
+
+    if len(found_signal_types) == 0:
```

Comment:
There is similar code before... would there be a point to extract this piece? [just a question]